app/src/main/java/org/oppia/android/app/options/AudioLanguageSelectionViewModel.kt

```diff
@@ -1,19 +1,110 @@
 package org.oppia.android.app.options
 
+import androidx.databinding.ObservableField
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
 import org.oppia.android.app.fragment.FragmentScope
+import org.oppia.android.app.model.AppLanguageSelection
 import org.oppia.android.app.model.AudioLanguage
+import org.oppia.android.app.model.OppiaLanguage
+import org.oppia.android.app.model.ProfileId
 import org.oppia.android.app.translation.AppLanguageResourceHandler
 import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.domain.oppialogger.OppiaLogger
+import org.oppia.android.domain.translation.TranslationController
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProvider
+import org.oppia.android.util.data.DataProviders.Companion.combineWith
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import org.oppia.android.util.locale.OppiaLocale
 import javax.inject.Inject
 
-/** Language list view model for the recycler view in [AudioLanguageFragment]. */
+private const val PRE_SELECTED_LANGUAGE_PROVIDER_ID = "systemLanguage+appLanguageProvider"
+
+/** ViewModel for managing language selection in [AudioLanguageFragment]. */
 @FragmentScope
 class AudioLanguageSelectionViewModel @Inject constructor(
   private val fragment: Fragment,
-  private val appLanguageResourceHandler: AppLanguageResourceHandler
+  private val appLanguageResourceHandler: AppLanguageResourceHandler,
+  private val translationController: TranslationController,
+  private val oppiaLogger: OppiaLogger
 ) : ObservableViewModel() {
+  private lateinit var profileId: ProfileId
+
+  /** An [ObservableField] to bind the resolved audio language to the dropdown text. */
+  val selectedAudioLanguage = ObservableField("")
+
+  private val appLanguageSelectionProvider: DataProvider<AppLanguageSelection> by lazy {
+    translationController.getAppLanguageSelection(profileId)
+  }
+
+  private val systemLanguageProvider: DataProvider<OppiaLocale.DisplayLocale> by lazy {
+    translationController.getSystemLanguageLocale()
+  }
+
+  private val languagePreselectionProvider: DataProvider<OppiaLanguage> by lazy {
+    appLanguageSelectionProvider.combineWith(
+      systemLanguageProvider,
+      PRE_SELECTED_LANGUAGE_PROVIDER_ID
+    ) { appLanguageSelection: AppLanguageSelection, displayLocale: OppiaLocale.DisplayLocale ->
+      val appLanguage = appLanguageSelection.selectedLanguage
+      val systemLanguage = displayLocale.getCurrentLanguage()
+      getPreselection(appLanguage, systemLanguage)
+    }
+  }
+
+  private fun getPreselection(
```

Comment:
Suggest: computePreselection, though this conflicts with the version below. Maybe rethink the helper names a bit.

Also here & elsewhere in the PR: please make note of member ordering. We generally keep fields together and methods together rather than interlacing them (since that can make it easier to find stuff). Separately, we usually keep public methods above private methods, though there's less consistency for public vs. private fields.

Reply:
I have also renamed the helper functions to avoid get/set as much as possible.